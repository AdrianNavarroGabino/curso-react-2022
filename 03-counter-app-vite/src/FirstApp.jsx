import PropTypes from "prop-types";

const newMessage = {
    message: 'Hola, mundo',
    title: 'Adrián'
};

const getResult = (a, b) => a + b;

export const FirstApp = ({title, subtitle}) => {
    return (
        <>
            <h1>{title}</h1>
            <code>{JSON.stringify(newMessage)}</code>
            <p>{getResult(3, 6)}</p>
            <p>Soy un subtítulo: {subtitle + 1}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 0
}