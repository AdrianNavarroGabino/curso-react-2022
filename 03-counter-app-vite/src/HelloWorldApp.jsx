export const HelloWorldApp = () => (<h1>Hola, mundo</h1>);