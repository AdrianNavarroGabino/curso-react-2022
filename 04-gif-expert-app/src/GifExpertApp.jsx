export const GifExpertApp = () => {
    return (
        <>
            {/* TÃ­tulo */}
            <h1>Gif Expert App</h1>

            {/* Input */}

            {/* Listado de Gif */}
            {/* Gif item */}
        </>
    )
}