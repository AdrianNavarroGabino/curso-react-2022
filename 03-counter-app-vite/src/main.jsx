import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import './styles.css';
import { CounterApp } from './CounterApp';

/*ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <FirstApp title="Hola, mundo" subtitle={123}/>
    </React.StrictMode>
)*/

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/*<CounterApp value={0}/>*/}
        <FirstApp />
    </React.StrictMode>
)