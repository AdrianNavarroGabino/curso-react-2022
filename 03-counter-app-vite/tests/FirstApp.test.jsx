/** @jest-environment jsdom */
import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {
    test('Debe hacer match con el snapshot', () => {
        const title = 'Hola, soy Vegeta';
        const { container } = render(<FirstApp title={title} subtitle="subtitulo" />);

        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el título', () => {
        const title = 'Hola, soy Vegeta';
        const { container, getByText, getByTestId } = render(<FirstApp title={title} subtitle="subtitulo" />);

        expect(getByText(title)).toBeTruthy();

        /*const h1 = container.querySelector('h1');

        expect(h1.innerHTML).toContain(title);*/
        expect(getByTestId('test-title')).toBeTruthy();
        expect(getByTestId('test-title').innerHTML).toBe(title);
    });

    test('Debe mostrar el subtítulo enviado por props', () => {
        const title = 'Hola, soy Goku';
        const subtitle = 'Soy un subtítulo';
        const { getAllByText } = render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );

        expect(getAllByText(subtitle).length).toBe(1);
    });
});