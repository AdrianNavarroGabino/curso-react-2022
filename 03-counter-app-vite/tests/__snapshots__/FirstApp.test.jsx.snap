// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Vegeta
  </h1>
  <p>
    9
  </p>
  <p>
    Soy un subt√≠tulo: 
    1
  </p>
</div>
`;
